lib/console.js

```diff
@@ -249,6 +268,104 @@ Console.prototype.groupEnd = function groupEnd() {
     this[kGroupIndent].slice(0, this[kGroupIndent].length - 2);
 };
 
+const keyKey = 'Key';
+const valuesKey = 'Values';
+const indexKey = '(index)';
+const iterKey = '(iteration index)';
+
+
+const isArray = (v) => ArrayIsArray(v) || isTypedArray(v) || isBuffer(v);
+const inspect = (v) => {
+  const opt = { depth: 0, maxArrayLength: 3 };
+  if (v !== null && typeof v === 'object' &&
+      !isArray(v) && ObjectKeys(v).length > 2)
+    opt.depth = -1;
+  return util.inspect(v, opt);
+};
+
+const getIndexArray = (n) => ArrayFrom({ length: n }, (_, i) => inspect(i));
+
+// https://console.spec.whatwg.org/#table
+Console.prototype.table = function(tabularData, properties) {
+  if (properties !== undefined && !ArrayIsArray(properties)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'properties',
+                               'array', properties);
+  }
+
+  if (tabularData == null ||
+      (typeof tabularData !== 'object' && typeof tabularData !== 'function') ||
+      (!isArray(tabularData) && ObjectKeys(tabularData) > 20))
+    return this.log(tabularData);
+
+  const final = (k, v) => this.log(cliTable(k, v));
+
+  const mapIter = isMapIterator(tabularData);
+  if (mapIter)
+    tabularData = previewMapIterator(tabularData);
+
+  if (mapIter || isMap(tabularData)) {
+    const keys = [];
+    const values = [];
+    for (const [k, v] of tabularData) {
+      keys.push(inspect(k));
+      values.push(inspect(v));
+    }
+    return final([
+      iterKey, keyKey, valuesKey
+    ], [
+      getIndexArray(tabularData.length || tabularData.size),
+      keys,
+      values,
+    ]);
+  }
+
+  const setIter = isSetIterator(tabularData);
+  if (setIter)
+    tabularData = previewSetIterator(tabularData);
+
+  if (setIter || isSet(tabularData)) {
+    const values = [];
+    for (const v of tabularData)
+      values.push(inspect(v));
+    return final([iterKey, valuesKey], [
+      getIndexArray(tabularData.length || tabularData.size),
+      values,
+    ]);
+  }
+
+  const map = {};
+  let hasPrimitives = false;
+  const valuesKeyArray = [];
+  const indexKeyArray = ObjectKeys(tabularData);
+
+  for (var i = 0; i < indexKeyArray.length; i++) {
+    const item = tabularData[indexKeyArray[i]];
+    if (properties === undefined && (item === null ||
+        (typeof item !== 'function' && typeof item !== 'object'))) {
+      hasPrimitives = true;
+      valuesKeyArray[i] = inspect(item);
```

Comment:
Would you be so kind and add a comment why a holey array is created here and below? (I see the point of it but it makes it easier when you read it for the first time to understand what is going on)